src/sonic-yang-models/yang-models/sonic-mirror-session.yang

```diff
@@ -0,0 +1,148 @@
+module sonic-mirror-session {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-mirror-session";
+    prefix sms;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-port {
+        prefix port;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    description
+        "SONiC Mirror session yang model";
+
+    revision 2021-06-15 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-mirror-session {
+
+        container MIRROR_SESSION {
+
+            list MIRROR_SESSION_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32 {
```

Comment:
Why is it max len 32, any backend restriction?

Reply:
No Backend limitation. But as discussed in meeting, keeping this check same.